### PR TITLE
more RunCommand initialization verification for TestStartBuildRunFollowLogs

### LINKED ISSUE #48: intermittent panics with TestStartBuildRunFollowLog
/kind bug

still seeing some intermittent panics with the multi-threaded `TestStartBuildRunFollowLog` unit test I recently added

```
=== RUN   TestStartBuildRunFollowLog
    run_test.go:173: Run initialization did not complete in time
--- FAIL: TestStartBuildRunFollowLog (3.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1b9e6cd]

goroutine 9 [running]:
testing.tRunner.func1.2({0x1c9cdc0, 0x2c0bd70})
	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1209 +0x36c
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1212 +0x3b6
panic({0x1c9cdc0, 0x2c0bd70})
	/opt/hostedtoolcache/go/1.17.1/x64/src/runtime/panic.go:1047 +0x266
github.com/shipwright-io/cli/pkg/shp/cmd/build.(*RunCommand).onEvent(0xc0001901c0, 0xc000346400)
	/home/runner/work/cli/cli/pkg/shp/cmd/build/run.go:128 +0xe8d
github.com/shipwright-io/cli/pkg/shp/cmd/build.TestStartBuildRunFollowLog(0xc0003e6b60)
	/home/runner/work/cli/cli/pkg/shp/cmd/build/run_test.go:178 +0x1508
testing.tRunner(0xc0003e6b60, 0x1ed3f18)
	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1259 +0x230
created by testing.(*T).Run
	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1306 +0x727
FAIL	github.com/shipwright-io/cli/pkg/shp/cmd/build	3.211s
```

### PR DESCRIPTION
**Pull request by gabemontero**
# Changes

Fixes #48 

/kind bug

/assign @coreydaley 
/assign @jkhelil

# Submitter Checklist

- [ /] Includes tests if functionality changed/was added
- [ n/a] Includes docs if changes are user-facing
- [ /] [Set a kind label on this PR](https://prow.k8s.io/command-help#kind)  
- [ /] Release notes block has been filled in, or marked NONE

See [the contributor guide](https://github.com/shipwright-io/build/blob/main/CONTRIBUTING.md)
for details on coding conventions, github and prow interactions, and the code review process.

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by gabemontero** [comment]
passed once ... rerun again to hopefully see consistency

**Comment by gabemontero** [comment]
passed again 

/assign @otaviof 

as well if he is interested 

but I'll take an lgtm from anybody :-)

File: pkg/shp/cmd/build/run_test.go [hunk 1]
```diff
@@ -162,7 +162,9 @@ func TestStartBuildRunFollowLog(t *testing.T) {
 		// the init to occur.
 		cmd.watchLock.Lock()
 		err := wait.PollImmediate(1*time.Second, 3*time.Second, func() (done bool, err error) {
-			if cmd.pw != nil {
+			// check any of the vars on RunCommand that are used in onEvent and make ser they are set;
```
Comment (coreydaley) [inline pkg/shp/cmd/build/run_test.go:165]:
```suggestion
			// check any of the vars on RunCommand that are used in onEvent and make sure they are set;
```

Comment (gabemontero) [reply pkg/shp/cmd/build/run_test.go:165]: thanks

**Comment by gabemontero** [comment]
spelling fix pushed @coreydaley thanks

**Review by coreydaley — APPROVED** [review]

**Pull request merged**